src/server/game/Conditions/ConditionMgr.cpp

```diff
@@ -2165,7 +2165,7 @@ bool ConditionMgr::isConditionTypeValid(Condition* cond)
             {
                 if (GameObjectData const* goData = sObjectMgr->GetGameObjectData(cond->ConditionValue3))
                 {
-                    if (cond->ConditionValue2 && goData->id != cond->ConditionValue2)
+                    if (cond->ConditionValue2 && goData->id1 != cond->ConditionValue2)
```

Comment:
?

Reply:
With this system, creatures and gob share the same spawning data structure. And with the malcrom PR about multiple creature ids, id = id1.